curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-asynchronous-programming-by-building-an-fcc-forum-leaderboard/645f7879ebbdb201892e55e1.md

```diff
@@ -26,7 +26,12 @@ assert(code.match(/const\s+userAvatarUrl\s*=/));
 You should assign a `ternary` operator to the `userAvatarUrl`. Your `ternary` should check if `avatar` starts with `/user_avatar/` and return `avatarUrl.concat(avatar)` or `avatar`.
 
 ```js
-assert(code.match(/const\s+userAvatarUrl\s*=\s*avatar\.startsWith\(\s*('|")\/user_avatar\/\1\s*\)\s*\?\s*avatarUrl\.concat\(\s*avatar\s*\)\s*:\s*avatar\s*/));
+const hasStartsWith = code.includes('startsWith');
+const regex = hasStartsWith
+   ? /const\s+userAvatarUrl\s*=\s*avatar\.startsWith\(\s*('|")\/user_avatar\/\1\s*\)\s*\?\s*avatarUrl\.concat\(\s*avatar\s*\)\s*:\s*avatar\s*/
+   : /const\s+userAvatarUrl\s*=\s*avatar\.test\(\s*\/(?:\^)?user_avatar\/\s*\)\s*\?\s*avatarUrl\.concat\(\s*avatar\s*\)\s*:\s*avatar\s*/;
+
+  assert(code.match(regex));
```

Comment:
```suggestion
assert(code.match(/const\s+userAvatarUrl\s*=\s*(?:avatar\.startsWith\(\s*('|")\/user_avatar\/\1\s*\)|\/\^\\\/user_avatar\\\/\/\.test\(\s*avatar\s*\))\s*\?\s*avatarUrl\.concat\(\s*avatar\s*\)\s*:\s*avatar/))
```
Condensed a bit, part that's differing between two cases might not be big enough to warrant two completely separate regexes.

`\/\^\\\/user_avatar\\\/\/` checks for `/^\/user_avatar\//`, `/` is part of the string that should be present to match the regex.